src/Enum.php

```diff
@@ -51,7 +52,7 @@ abstract class Enum implements \JsonSerializable
      * @psalm-pure
      * @param mixed $value
      *
-     * @psalm-param static<T>|T $value
+     * @psalm-param T $value
```

Comment:
This was changed because:

 1. psalm does not understand this union type
 2. `new MyEnum(new MyEnum('foo'))` makes no sense anyway

Reply:
`new MyEnum(new MyEnum('foo'))` is a supported use case though 🤔 

But your change only prevents that from happening explicitly, through static analysis. So after all this change makes sense I think.